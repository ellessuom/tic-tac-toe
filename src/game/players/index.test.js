import React from 'react';
import { shallow } from 'enzyme';
import PlayersParent from './index';

describe('<PlayersParent />', () => {
  it('should render layout structure correctly', () => {
    const component = shallow(<PlayersParent />);

    expect(component).toMatchSnapshot();
  });

  describe('<PlayersParent />\'s children', () => {
    let wrapper;
    beforeAll(() => {
      wrapper = shallow((<PlayersParent />));
    });

    it('should render both children', () => {
      expect(wrapper.children()).toHaveLength(2);
    });

    it('should render first child with correct playerId prop', () => {
      expect(wrapper.childAt(0).props().playerId).toBe('p1');
    });

    it('should render second child with correct playerId prop', () => {
      expect(wrapper.childAt(1).props().playerId).toBe('p2');
    });
  });
});
