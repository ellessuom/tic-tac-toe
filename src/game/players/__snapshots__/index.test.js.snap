// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayersParent /> should render layout structure correctly 1`] = `
<div
  className="players-wrapper"
>
  <BasePlayer
    playerId="p1"
  />
  <BasePlayer
    playerId="p2"
  />
</div>
`;
